Sources/Core/AWSClientRuntime/EventStream/AWSMessageSigner.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+
+extension AWSEventStream {
+    /// Signs a `Message` using the AWS SigV4 signing algorithm
+    public class AWSMessageSigner: MessageSigner {
+        let context: HttpContext
```

Comment:
I don't love the needing this dependency, can we abstract this away perhaps in a closure?

```swift
let signingConfig: () async throws -> AWSSigningConfig
```

Reply:
Additionally, can we just inject `previousSignature` as well?

Reply:
I was thinking on this in the shower, we want to delay the signing until the request signature is calculated.

Hence, we need the HttpContext as injected dependency than plan signature. if we pass plain string, it must be computed at the construction time which is not going to work here. Rather it must be calculated when first message is queried.